src/Credentials/ServiceAccountJwtAccessCredentials.php

```diff
@@ -128,10 +132,17 @@ public function updateMetadata(
     public function fetchAuthToken(callable $httpHandler = null)
     {
         $audience = $this->auth->getAudience();
-        if (empty($audience)) {
+        $scope = $this->auth->getScope();
+        if (empty($audience) && empty($scope)) {
             return null;
         }
 
+        if (!empty($audience) && !empty($scope)) {
+            throw new UnexpectedValueException(
```

Comment:
```suggestion
            throw new \UnexpectedValueException(
```

Would we be able to document this being thrown? Wondering if we need a test for this as well.

Reply:
Good catch and yes, I added a test!

I'm happy to add `@throws`. In this library in general (not that it's the right thing to do), we haven't been documenting thrown exceptions.

Reply:
I trust your judgement 👍